mobile/App.tsx

```diff
@@ -1,43 +1,81 @@
 import { WebView, WebViewMessageEvent } from 'react-native-webview';
 import { SafeAreaView, Text, Platform } from 'react-native';
 import { torClient } from './services/Tor';
+import Clipboard from '@react-native-clipboard/clipboard';
 
 const App = () => {
   const webViewRef = useRef<WebView>();
   const uri = (Platform.OS === 'android' ? 'file:///android_asset/' : '') + 'Web.bundle/index.html';
 
-  const injectMessage = (id: string, data: object) => {
+  const injectMessageResolve = (id: string, data: object) => {
     const json = JSON.stringify(data);
     webViewRef.current?.injectJavaScript(
       `(function() {window.NativeRobosats.onMessageResolve(${id}, ${json});})();`,
     );
   };
 
+  const injectMessage = (message: object) => {
+    const json = JSON.stringify(message);
+    webViewRef.current?.injectJavaScript(
+      `(function() {window.NativeRobosats?.onMessage(${json});})();`,
+    );
+  };
+
   const onMessage = async (event: WebViewMessageEvent) => {
     const data = JSON.parse(event.nativeEvent.data);
     if (data.category === 'http') {
+      sendTorStatus();
+
       if (data.type === 'get') {
-        torClient.get(data.path).then((response: object) => {
-          injectMessage(data.id, response);
-        });
+        torClient
+          .get(data.path)
+          .then((response: object) => {
+            injectMessageResolve(data.id, response);
+          })
+          .finally(sendTorStatus);
       } else if (data.type === 'post') {
-        torClient.post(data.path, data.body, data.headers).then((response: object) => {
-          injectMessage(data.id, response);
-        });
+        torClient
+          .post(data.path, data.body, data.headers)
+          .then((response: object) => {
+            injectMessageResolve(data.id, response);
+          })
+          .finally(sendTorStatus);
       } else if (data.type === 'delete') {
-        torClient.delete(data.path, data.headers).then((response: object) => {
-          injectMessage(data.id, response);
-        });
+        torClient
+          .delete(data.path, data.headers)
+          .then((response: object) => {
+            injectMessageResolve(data.id, response);
+          })
+          .finally(sendTorStatus);
       } else if (data.type === 'xhr') {
-        torClient.request(data.path).then((response: object) => {
-          injectMessage(data.id, response);
-        });
+        torClient
+          .request(data.path)
+          .then((response: object) => {
+            injectMessageResolve(data.id, response);
+          })
+          .finally(sendTorStatus);
+      }
+    } else if (data.category === 'system') {
+      if (data.type === 'copyToClipboardString') {
+        Clipboard.setString(data.detail);
       }
     }
   };
 
-  torClient.startDaemon();
+  const sendTorStatus = async () => {
+    let daemonStatus;
+    try {
+      daemonStatus = await torClient.daemon.getDaemonStatus();
+    } catch {
+      daemonStatus = 'ERROR';
```

Comment:
Apparently not working. I simply opened the app without internet connection. Saw the orange Tor icon until I got this unhandled error. Hope it helps:
![3](https://user-images.githubusercontent.com/90936742/194297796-203709ee-4c45-4bdc-8268-e687eb35ad38.png)
![1](https://user-images.githubusercontent.com/90936742/194297808-fb1f2fd8-9fdf-40fd-a9ee-37d1fd1f3168.png)

Reply:
Looks like the Daemon can't manage network issues out of Tor specific issues, it sounds like if want to proceed with that we'll need to implement a Network checker from the android side, sdo you think it's worth it?

Reply:
> do you think it's worth it?

Ummm, ideally the app should gracefully fail if the device has no internet. Probably worth it!